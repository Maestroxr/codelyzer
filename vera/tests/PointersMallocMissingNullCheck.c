
int main() 
{
	int returnValue = 0;
	
	/*OK*/
	int* p1 = (int*)malloc(sizeof(int));
	if (!p1)
	{
		returnValue = -1;
	}
	int* p2 = (int*)malloc(sizeof(int));
	if (p2 != NULL)
	{
		returnValue = 2;
	}
	p2 = (int*)malloc(sizeof(int));
	if (NULL != p2)
	{
		returnValue = 2;
	}
	p1 = (int*)malloc(sizeof(int));
	if (NULL == p1)
	{
		returnValue = -1;
	}
	int* p3 = (int*)malloc(sizeof(int));
	if (p3 == NULL)
	{
		returnValue = -12;
	}
	/*--------*/
	
	/*PROBLEM*/
	
	if (!p1)
	{
		returnValue = -1;
	}
	p1 = (int*)malloc(sizeof(int));
	
	
	p2 = (int*)malloc(sizeof(int));
	if (p3 != NULL)
	{
		returnValue = 2;
	}
	
	p3 = (int*)malloc(sizeof(int));
	p3 = (int*)malloc(sizeof(int));
	if (p3 == NULL)
	{
		returnValue = -12;
	}
	
	int* p4 = (int*)malloc(sizeof(int));

	/*********/
	
}


