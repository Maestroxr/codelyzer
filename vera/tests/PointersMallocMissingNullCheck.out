C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocMissingNullCheck.c:36: Pointer's 'problem1' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocMissingNullCheck.c:42: Pointer's 'problem2' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocMissingNullCheck.c:47: Pointer's 'problem3' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocMissingNullCheck.c:53: Pointer's 'problem5' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocMissingNullCheck.c:53: Pointer's 'problem4' malloc is missing null pointer check.
